server/text_generation_server/layers/attention/rocm.py

```diff
@@ -227,29 +225,32 @@ def paged_attention(
 
 
 SUPPORTS_WINDOWING = False
-if ENGINE == "ck":
-
-    def attention(
-        q,
-        key_cache: torch.Tensor,
-        value_cache: torch.Tensor,
-        seqlen: Seqlen,
-        block_tables: torch.Tensor,
-        softmax_scale: float,
-        window_size_left: int = -1,
-        causal: bool = True,
-        softcap: float = 0.0,
-    ):
+
+
+def attention(
+    *query,
```

Comment:
Isn't that supposed to be `*, query, ` ? 

Also it doesn't matter as much since we force kwargs, but key, value, key_cache, value_cache, somehow makes more sense to my brain (same tensor shapes, come from the same location).

Nit really

Reply:
> Isn't that supposed to be `*, query,`  ?

Indeed, fixed!

Reply:
> Also it doesn't matter as much since we force kwargs, but key, value, key_cache, value_cache, somehow makes more sense to my brain (same tensor shapes, come from the same location).

I think now that the KV cache is a separate data type, we should just pass that in the future.

Reply:
Also done.